File: 17_repeat_string.js
// repeat a string num times

function repeatStringNumTimes(str, num){
       if (num < 0) return "";   
       return str.repeat(num);
}


// without using built-in function
function repeatStringForLoop(str, num){
    if (num < 0) return "";
    final = "";
    for(var i=0; i <num; i++){
        final += str;
    }
    return final;
}

// using recuision method
function repeatStringRecuision(str, num){
    if(num < 0) return "";

    if(num ===1) return str;
    return str + repeatStringRecuision(str, num - 1);
}

console.log(repeatStringNumTimes("abc", 3));
console.log(repeatStringNumTimes("abc", 4));
console.log(repeatStringRecuision("a bc", 5));

File: 17_repeat_string.py
# repeat a string num times

def repeatStringNumTimes(string, num):
    if num < 0:
        return ""
    return string * num


# using for loop
def repeatStringForLoop(string, num):
    if num < 0:
        return ""
    final = ""
    for i in range(num):
         final += string
    return final

# using recuision method
def repeatStringRecuision(string, num):
    if num < 0:
        return ""
    elif num == 1:
        return string
    else:
        return string + repeatStringRecuision(string, num-1)


print(repeatStringNumTimes("abc", 3))
print(repeatStringForLoop("abc", 4))
print(repeatStringRecuision("a bc", 5))